lib/pure/collections/sets.nim

```diff
@@ -80,6 +80,8 @@ type
     ## <#initOrderedSet,int>`_ before calling other procs on it.
     data: OrderedKeyValuePairSeq[A]
     counter, first, last: int
+  AnySet1[T] = HashSet[T] | OrderedSet[T]
```

Comment:
- Is it needed to have both AnySet1 and AnySet2? Why not use the same type for both args?
- Also, the convention is to use SomeFoo. E.g. we have SomeNumber, SomeInteger. So may be use SomeSet here?

Reply:
* It is needed because the sets can be different, but not mandatory different. I tried `proc incl*[T](s: var AnySet[T], other: distinct AnySet[T])`, but unfortunately for some reason this requires mandatory different sets. I'm not sure whether this is by specification or it is a bug in the compiler.

* I will change it to `SomeSet1` and `SomeSet2`.

Reply:
Thanks. May be someone else can comment if it should be possible to use to the same type for the `incl`.